cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,922 @@
+    }
+    assert(chosen_elem_per_thread > 0);
+    assert(chosen_tile_size > 0);
+    assert((sizeof...(RandomAccessIteratorsIn) == 0) != (chosen_smem_size != 0)); // logical xor
+
+    const Offset grid_dim = ::cuda::ceil_div(num_items, Offset{chosen_tile_size});
+    return ::cuda::std::make_tuple(
+      THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron(grid_dim, block_dim, chosen_smem_size, stream),
+      kernel,
+      chosen_elem_per_thread);
+  }
+
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t
+  invoke_algorithm(cuda::std::index_sequence<Is...> is, ::cuda::std::integral_constant<Algorithm, Algorithm::ublkcp>)
+  {
+    auto ret = configure_ublkcp_kernel<ActivePolicy>(is);
+    if (!ret)
+    {
+      return ret.error;
+    }
+    // TODO(bgruber): use a structured binding in C++17
+    // auto [launcher, kernel, elem_per_thread] = *ret;
+    return ::cuda::std::get<0>(*ret).doit(
+      ::cuda::std::get<1>(*ret),
+      num_items,
+      ::cuda::std::get<2>(*ret),
+      op,
+      out,
+      make_ublkcp_ptr_set<const value_t<RandomAccessIteratorsIn>>(
+        THRUST_NS_QUALIFIER::unwrap_contiguous_iterator(::cuda::std::get<Is>(in)))...);
+  }
+
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t invoke_algorithm(
+    cuda::std::index_sequence<Is...> is, ::cuda::std::integral_constant<Algorithm, Algorithm::memcpy_async>)
+  {
+    auto ret = configure_memcpy_async_kernel<ActivePolicy>(is);
+    if (!ret)
+    {
+      return ret.error;
+    }
+    // TODO(bgruber): use a structured binding in C++17
+    // auto [launcher, kernel, elem_per_thread] = *ret;
+    return ::cuda::std::get<0>(*ret).doit(
+      ::cuda::std::get<1>(*ret),
+      num_items,
+      ::cuda::std::get<2>(*ret),
+      op,
+      out,
+      THRUST_NS_QUALIFIER::unwrap_contiguous_iterator(::cuda::std::get<Is>(in))...);
+  }
+
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t invoke_algorithm(
+    cuda::std::index_sequence<Is...>, ::cuda::std::integral_constant<Algorithm, Algorithm::unrolled_staged>)
+  {
+    using policy_t              = typename ActivePolicy::unrolled_policy;
+    constexpr int block_dim     = policy_t::BLOCK_THREADS;
+    constexpr int max_occupancy = 8; // TODO(bgruber): guestimate for A6000, where to get this from?
+#if _CCCL_STD_VER >= 2014
+    constexpr int num_elem_per_thread =
+      ::cuda::ceil_div(ActivePolicy::min_bif, max_occupancy * block_dim * loaded_bytes_per_iter);
+#else
+    constexpr int mult                = max_occupancy * block_dim * loaded_bytes_per_iter;
+    constexpr int num_elem_per_thread = (ActivePolicy::min_bif + mult - 1) / mult;
+#endif
+
+#if _CCCL_STD_VER >= 2014
+    constexpr int num_elem_per_thread_clamped =
+      ::cuda::std::clamp(num_elem_per_thread, +policy_t::MIN_ITEMS_PER_THREAD, +policy_t::MAX_ITEMS_PER_THREAD);
```

Comment:
question: I see unary `+` in many places, what's the intention?

Reply:
`clamp` takes arguments by `const T&` so it ODR-uses the constants `policy_t::MIN_ITEMS_PER_THREAD`, which nvcc does not like. The unary-plus creates a copy of the constant value, which is passed to `clamp` instead,  as a workaround.